sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -174,6 +197,54 @@ void printArgs(Arg0 arg0, Args... args) {
   pi::printArgs(std::forward<Args>(args)...);
 }
 
+template <typename T> struct printOut {
+  printOut(T val) {}
+}; // Do nothing
+
+template <> struct printOut<PiEvent *> {
+  printOut(PiEvent *val) {
+    std::cout << std::endl << "       ";
+    std::cout << "[out]pi_event * : " << val;
+    if (val) {
+      std::cout << "[ " << *val << " ... ]";
+    }
+  }
+};
+
+template <> struct printOut<PiMem *> {
+  printOut(PiMem *val) {
+    std::cout << std::endl << "       ";
+    std::cout << "[out]pi_mem * : " << val;
```

Comment:
```suggestion
    std::cout << "\t[out]pi_mem * : " << val;
```

Reply:
Thank you, fixed in all relevant places.